### PR TITLE
Separate compiler and validator with virtualenv

### PR DESCRIPTION
**Pull request by maany**

### CODE REVIEW

File: simple [hunk 1]
```diff
@@ -69,26 +76,31 @@ create-template)
...
+	    virtualenv $compiler_env
+	    source $compiler_env/bin/activate && pip install simple-grid-yaml-compiler && touch $installed_compiler
 	) || exit
 
+	cd $compiler_dir
+	source $compiler_env/bin/activate
 	simple_grid_yaml_compiler $conf -o output.yaml -s schema.yaml
 	val=$?
 
 	deactivate
```
Comment (maarten-litmaath) [inline simple:103]: Is that deactivate still needed right before the subshell exits?

File: simple [hunk 2]
```diff
@@ -220,16 +232,46 @@ docker)
...
+        ) || exit
+
+        [ -e $installed_virtualenv ] || (
+            yum install -y python-virtualenv && touch $installed_virtualenv
+        ) || exit
+
+	[ -e $installed_git ] || (
+	    yum install -y git && touch $installed_git
+	) || exit
+	echo $validator_dir
```
Comment (maarten-litmaath) [inline simple]: why is that echo needed?

File: simple [hunk 3]
```diff
@@ -220,16 +232,46 @@ docker)
...
+        [ -e $installed_validator ] || (
+            mkdir -p $validator_dir &&  cd $validator_dir && pwd &&
```
Comment (maarten-litmaath) [inline simple]: why the pwd?

File: simple [hunk 4]
```diff
@@ -220,16 +232,46 @@ docker)
...
+	    git clone "$validator_url" $validator_dir &&
+	    virtualenv $validator_env && source $validator_env/bin/activate &&
+	    pip install -r requirements.txt && pip install $validator_dir &&
```
Comment (maarten-litmaath) [inline simple:257]: is the argument to the second pip install correct?

File: simple [hunk 5]
```diff
@@ -220,16 +232,46 @@ docker)
...
+	    deactivate && touch $installed_validator
```
Comment (maarten-litmaath) [inline simple:258]: is the deactivate still needed?

File: simple (hunk 4 above)
Comment (maany) [reply simple:257]: Yup, the first one installs the dependencies while the second one installs the validation engine itself.

File: simple (hunk 2 above)
Comment (maany) [reply simple]: Not needed. I was debugging.

File: simple (hunk 3 above)
Comment (maany) [reply simple]: I'll get rid of it. Used it for debugging.

File: simple (hunk 4 above)
Comment (maarten-litmaath) [reply simple:257]: The 2nd pip install has a _directory_ as an argument, instead of a product...

File: simple (hunk 5 above)
Comment (maany) [reply simple:258]: Yup, the validation engine is installed in the currently active $validator_env (line 256) which needs to be deactivated after the installation.
For the YAML compiler and the validation engine, the installation steps are as follows:
1. Create a virtualenv
1. activate the virtualenv
1. install the package (compiler through PyPI, validator through source code)
1. deactivate

The steps to use them are:

1. activate the respective virtualenv
1. issue the command line instruction to execute the compiler or the validator
1. deactivate the respective virtualenv

File: simple (hunk 4 above)
Comment (maany) [reply simple:257]: Yup, that is intentional. Pip can install packages from source code as well. The directory contains the product i.e. the source of the infra validation engine python package which is not available on PyPi as I have not released it. Therefore, I clone the code from GitHub and install the package with pip as seen above.

File: simple (hunk 1 above)
Comment (maany) [reply simple:103]: Ah I see your point. Maybe not. I can try without ;)

File: simple (hunk 5 above)
Comment (maarten-litmaath) [reply simple:258]: My question is really: is a deactivate command needed right before a shell exits?

Comment (maany) [reply simple:258]: Nope, you're right. I see your point. It is not needed. ;)

File: simple (hunk 1 above)
Comment (maarten-litmaath) [reply simple:103]: Or must we not make any assumptions about what a deactivation may or may not do?

Comment (maany) [reply simple:103]: Seems to work okay. I have removed it for now ;)

**Comment by maany** [comment]
@maarten-litmaath thanks for the review. I've made the changes. Merging this PR now.

**Pull request merged**